.github/workflows/publish.yml

```diff
@@ -0,0 +1,47 @@
+name: Publish
+on:
+    release:
+        types: [published]
+
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Xero-Java repo
+              uses: actions/checkout@v4
+              with:
+                repository: XeroAPI/Xero-Java
+                path: Xero-Java
+
+            - name: Set up JDK environment
+              uses: actions/setup-java@v4
+              with: 
+                distribution: 'temurin'
+                java-version: '11'
+                cache: maven
+
+            - name: Import GPG Key
+              run: |
+                echo "${{ secrets.GPG_PRIVATE_KEY }}" | gpg --batch --import
+              env: 
+                GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY}}
+
+            - name: Configure Maven settings
+              run: |
+                mkdir -p ~/.m2
+                echo "<settings>
```

Comment:
https://github.com/actions/setup-java has option to set mvn serverid, user, password. That seems much cleaner

Reply:
refactored the logic as per the review comments